site/docs/skylark/aspects.md

```diff
@@ -75,38 +75,72 @@ example. An aspect implementation function is then invoked on all nodes in
 the shadow graph similar to how rule implementations are invoked on the nodes
 of the original graph.
 
-## Defining aspects
+## Simple example
 
-Aspect definitions are similar to rule definitions, and defined using
-the [`aspect`](lib/globals.html#aspect) function. Let's take a look at
-the example:
+This example demonstrates how to recursively print the source files for a 
+rule and all of its  dependencies that have a `deps` attribute. It shows 
+how to define an empty provider, an aspect implementation and an aspect 
+definition. It invokes the aspect from the bazel command line.
 
 ```python
-metal_proto_aspect = aspect(implementation = _metal_proto_aspect_impl,
+# Our rule returns nothing so generate an empty provider.
+PrintProvider = provider()
+
+def _print_aspect_impl(target, ctx):
+    # Make sure the rule has a srcs attribute.
+    if hasattr(ctx.rule.attr, "srcs"):
+        # Iterate through the files that make up the sources and 
+        # print their paths.
+        sources = [f for src in ctx.rule.attr.srcs for f in src.files]
```

Comment:
Please unroll this into loops or multiple comprehensions

Reply:
Fixed

Reply:
fixed